src/Access/AccessControl.h

```diff
@@ -169,6 +169,7 @@ class AccessControl : public MultipleAccessStorage
     std::shared_ptr<const ContextAccess> getContextAccess(
         const UUID & user_id,
         const std::vector<UUID> & current_roles,
+        const std::vector<UUID> & additional_roles,
```

Comment:
Why can't we combine those `additional_roles` with `current_roles` and use one parameter instead of two ones?

Reply:
Because there is also `use_default_roles`, that controls which roles are applied to the `ContextAccess`.

I'm not entirely sure why `use_default_roles` should be `false` or `true`, hence wasn't sure if the same logic should be applied to `external_roles`.